homepage/tests/my_profile_tests.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+from pytest_django.asserts import assertTemplateUsed
+
+
+@pytest.mark.django_db
+class TestMyProfile:
+
+    @pytest.fixture
+    def signed_up_user_details(self):
+        return {'username': 'testUser1', 'password': 'password123'}
+
+    def test_user_renders_profile_template(self, client, signed_up_user_details):
+        # log in with testUser1 using the sign-in page to acesss his profile
+        client.post('/users/sign_in/', data=signed_up_user_details)
+
+        response = client.get('/users/my_profile/')
+        assert response.status_code == 200
+        assertTemplateUsed(response, 'homepage/users/my_profile.html')
+
+    # AnonymousUser is Django's request.user when no user is currently logged in
+    def test_anonymous_user_accesses_my_profile(self, client):
+        try:
+            client.get('/users/my_profile/')
+            valid_access = True
+        except AttributeError:
+            valid_access = False
+
+        assert not valid_access

```

Comment:
Should have tests for display of followed courses and reviews:
- if user has them - that the ones displayed do belong to him
- if user doesn't have them - that none appear

Reply:
@VeraTs 

Do you know how do i get the followed courses and the reviews from the HTML page?
I wanted to add some tests but i wasn't sure how to get the matching elements from the HTML page

Reply:
yeah;
first, notice in the line I marked in views.py, that you pass the followed courses and reviews to the profile page when you render it.
When in the test file, you get the request, it comes with a context attribute, which is a dictionary, and if you do, for example:
`request.context['user_reviews'] You'll get the list that is passed to the page when it's rendered